pkg/storage/tsdb/block/index.go

```diff
@@ -757,6 +768,40 @@ func rewrite(
 	return nil
 }
 
+// clampChunk returns a chunk that is clamped by the minT and maxT so that all samples fall within these,
+// else returns nil if no chunk clamping was needed.
+func clampChunk(input *chunks.Meta, minT, maxT int64) (*chunks.Meta, error) {
```

Comment:
(Feel free to ignore) There could be top-checks here to avoid re-encoding inside chunks or iterating through outside chunks.
- `input.NumSamples() <= 0 || input.MinTime >= maxT || input.MaxTime < minT` for empty/outside chunks to return `nil, nil`
- `minT <= input.MinTime && input.MaxTime < maxt` to return `input, nil` for inside chunks.